### PR TITLE
[APP-657] Add share list functionality

### PR DESCRIPTION
**Pull request by ansh**
- [x] Consolidate list actions into a component
- [x] Fix styling on web
- [x] Refactor and clean up dead code/styles
- [x] Add button to share list
- [x] Setup embed view for shared list ~~(currently BLOCKED)~~ no longer blocked, thanks @dholms 

To test, make sure you are on the `ansh/list-embed` branch from `atproto`

### CODE REVIEW

**Comment by ansh** [comment]
Testing failing due to regression on main branch

File: src/state/models/content/list.ts [hunk 1]
```diff
@@ -97,6 +97,10 @@ export class ListModel {
     return this.list?.creator.did === this.rootStore.me.did
   }
 
+  get isSubscribed() {
+    return this.list?.viewer?.muted
```
[the remark this answers is not in the document]
Comment (pfrazee) [reply src/state/models/content/list.ts:101]: Oh it might be, right. Just double check me on this.

**Review by pfrazee — APPROVED** [review]
Okay this is great. Good stuff

File: src/state/models/content/list.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (ansh) [reply src/state/models/content/list.ts:101]: yeah for some reason it’s called muted in the lexicon but what it’s really trying to tell us if the user is subscribed to the list or not. the other code also treats it as "subscribed" and puts it in a variable within the components so this getter follows convention.

**Pull request merged**